src/rust/src/pkcs7.rs

```diff
@@ -164,6 +166,257 @@ fn encrypt_and_serialize<'p>(
+                .encrypted_content_info
+                .encrypted_content
+                .unwrap();
+            let decrypted_content = symmetric_decrypt(py, algorithm, mode, encrypted_content)?;
+            pyo3::types::PyBytes::new(py, decrypted_content.as_slice())
+        }
+        _ => {
+            return Err(CryptographyError::from(
+                pyo3::exceptions::PyValueError::new_err(
+                    "The PKCS7 data is not an EnvelopedData structure.",
+                ),
+            ));
+        }
+    };
+
+    // If text_mode, strip the header if possible, else return an error
+    let plain_data = plain_content.as_bytes();
+    let plain_data = if options.contains(types::PKCS7_TEXT.get(py)?)? {
+        let header = b"Content-Type: text/plain\r\n\r\n";
+        if plain_data.starts_with(header) {
+            &plain_data[header.len()..]
+        } else {
+            return Err(CryptographyError::from(
+                pyo3::exceptions::PyValueError::new_err(
+                    "No 'Content-Type' header to be deleted in the decrypted data. Please remove the 'text' option.",
+                ),
+            ));
+        }
+    } else {
+        plain_data
+    };
+
+    let plain_data = pyo3::types::PyBytes::new(py, plain_data);
+    Ok(plain_data)
+}
+
+fn check_decrypt_parameters<'p>(
+    py: pyo3::Python<'p>,
+    certificate: &pyo3::Bound<'p, x509::certificate::Certificate>,
+    options: &pyo3::Bound<'p, pyo3::types::PyList>,
+) -> Result<(), CryptographyError> {
+    // Check if RSA encryption with PKCS1 v1.5 padding is supported (dependent of FIPS mode)
+    if cryptography_openssl::fips::is_enabled() {
+        return Err(CryptographyError::from(
+            exceptions::UnsupportedAlgorithm::new_err((
+                "RSA with PKCS1 v1.5 padding is not supported by this version of OpenSSL.",
+                exceptions::Reasons::UNSUPPORTED_PADDING,
+            )),
+        ));
+    }
+
+    // Check if all options are from the PKCS7Options enum
+    let pkcs7_option_type = types::PKCS7_TEXT.get(py)?.get_type();
+    if !options
+        .iter()
+        .all(|opt| opt.get_type().eq(pkcs7_option_type.clone()).unwrap())
+    {
+        return Err(CryptographyError::from(
+            pyo3::exceptions::PyValueError::new_err("options must be from the PKCS7Options enum"),
+        ));
+    }
+
+    // Check if any option is not PKCS7Options::Text
+    let text_option = types::PKCS7_TEXT.get(py)?;
+    if options
+        .iter()
+        .any(|opt| !opt.eq(text_option.clone()).unwrap())
+    {
+        return Err(CryptographyError::from(
+            pyo3::exceptions::PyValueError::new_err(
+                "Only the following options are supported for decryption: Text",
+            ),
+        ));
+    }
+
+    // Check if certificate's public key is an RSA public key
+    if !certificate
+        .call_method0(pyo3::intern!(py, "public_key"))?
+        .is_instance_of::<backend::rsa::RsaPublicKey>()
```

Comment:
Similar note to above, you need to use `types::RSA_PUBLIC_KEY` here.

Reply:
Done!